### ISSUE TITLE
Introduce optional HTTP proxy

### ISSUE THREAD

**Issue by kelson42** [openzim/mwoffliner#1171]
Most of the time, MWoffliner creates many ZIM files in a one run. Therefore, often, the same articles are scraped many times for many ZIM files.

Dealing with the upstream backend many times with the very same HTTP requests for the very same content is not the most performant approach. As a consequence a few strategies have been elaborated to avoid that:
* There is a local download cache configurable, which for many requests save/check against a local directory where files (and HTTP headers) are backuped. This is a really custom solution, it works, but cumbersome to maintain and probably quite slow considering that the way to handle the fs backend is super primitive.
* We have the idea to write many ZIM files [in parallel](https://github.com/openzim/mwoffliner/issues/701). This is theoritically a good idea but (1) this needs a new dev which is not trivial (2) This makes the MWoffliner even more complex and I'm not sure this is a good idea considering this is already complex (3) the memory consumption will be higher.

For these reason, maybe an other approach would be better: removing this MWoffliner local cache and just deal with a local HTTP proxy.

Such a proxy might be done at Docker level and should probably be optional (no reason to have it in many scraping scenarios). It could/should ideally be transparent so there nothing to change in MWoffliner configuration/code. We would not really benefit of it for other scrapers for the moment, but I believe this might be useful for the Zimit scraper as well and potentially other.

In term of software I came through a few of them. Nuster (based on HAproxy) seems quite intersting https://github.com/jiangwenyuan/nuster.

What do you think?

**Comment by kelson42** [comment]
All of this seems too complicated to me meanwhile for the potential benefit. I will close as WONTFIX.
